fy22_ferry_study/attraction_travel_times/distance_direction_parser.py

```diff
@@ -0,0 +1,164 @@
+            units="imperial",
+            departure_time=now,
+        )
+        names = {"name": name}
+        output.insert(0, names)
+
+        if mode == "driving":
+            driving_list.append(output)
+
+        if mode == "transit":
+            transit_list.append(output)
+
+    """loops through the destinations file and runs the distance duration function"""
+    for idx, row in destinations_df.iterrows():
+        lat = row["Latitude"]
+        lon = row["Longitude"]
+        name = row["Name"]
+        destination = (lat, lon)
+        distance_duration(destination)
+
+    if mode == "driving":
+        print("returning driving list")
+        return driving_list
+    if mode == "transit":
+        print("returning transit list")
+        return transit_list
+
+
+def save_list_in_memory(transpo_list):
+    """saves list from distance_duration_iteration into memory to avoid repeated calls to the API"""
+    transpo_list = transpo_list
+    return transpo_list
+
+
+def unpack_dicts(driving_list, transit_list):
+    """function to crack into the nested dictionary structure that the google api returns"""
+    df = pd.DataFrame()
+    transit_list_of_dicts = []
+    driving_list_of_dicts = []
+
+    for item in driving_list:
+        destination_lat = item[1]["legs"][0]["end_location"]["lat"]
+        destination_lng = item[1]["legs"][0]["end_location"]["lng"]
+        distance = item[1]["legs"][0]["distance"]["text"]
+        duration = item[1]["legs"][0]["duration"]["text"]
+        name = item[0]["name"]
+        d = {
+            "name": [name],
+            "lat": [destination_lat],
+            "lng": [destination_lng],
+            "d_distance": [distance],
+            "d_duration": [duration],
+        }
+        driving_list_of_dicts.append(d)
+
+    for item in transit_list:
+        if len(item) == 2:
+            destination_lat = item[1]["legs"][0]["end_location"]["lat"]
+            destination_lng = item[1]["legs"][0]["end_location"]["lng"]
+            distance = item[1]["legs"][0]["distance"]["text"]
+            duration = item[1]["legs"][0]["duration"]["text"]
+            name = item[0]["name"]
+            d = {
+                "t_distance": [distance],
+                "t_duration": [duration],
+            }
+            transit_list_of_dicts.append(d)
+        else:
+            d = {
+                "t_distance": 0,
+                "t_duration": 0,
+            }
+            transit_list_of_dicts.append(d)
+
+    df1 = pd.DataFrame(driving_list_of_dicts)
+    df2 = pd.DataFrame(transit_list_of_dicts)
+    frames = [df1, df2]
+    df = pd.concat(frames, axis=1)
+    return df
+
+
+def df_to_csv(df):
+    df.to_csv(GDRIVE_FOLDER / "attraction_travel_times.csv", sep=",")
+
+
+def unpack_geometries(transpo_list):
+    """accepts either the transit list or the driving list from the distance_duration_iteration function and parses accordingly"""
+    polylines = []
+
+    if transpo_list == driving_list:
```

Comment:
Functions should ideally only utilize variables that are passed in or created internally within the function. Looks like it's grabbing a global `driving_list` and `transit_list`. Although this works, it can potentially create some weird scoping issues. Just make these arguments to the function if you need to use them in here.